go/pkg/digest/digest.go

```diff
@@ -137,7 +137,12 @@ func NewFromString(s string) (Digest, error) {
 	if err != nil {
 		return Empty, fmt.Errorf("invalid size in digest %s: %s", s, err)
 	}
-	return New(pair[0], size)
+	if dg, err := New(pair[0], size); err != nil {
```

Comment:
Looks like `New` and `NewFromProto` would still return an invalid digest with an error. Can you update them as well? Or maybe just those two since this one is delegating to `New` anyways.